OptimizelySDK.Tests/DecisionServiceTest.cs

```diff
@@ -46,20 +46,19 @@ public class DecisionServiceTest
         [SetUp]
         public void SetUp()
         {
-            LoggerMock = new Mock<ILogger>();
-            ErrorHandlerMock = new Mock<IErrorHandler>();
-            UserProfileServiceMock = new Mock<UserProfileService>();
-            BucketerMock = new Mock<Bucketer>(LoggerMock.Object);
+            LoggerMock              = new Mock<ILogger>();
+            ErrorHandlerMock        = new Mock<IErrorHandler>();
+            UserProfileServiceMock  = new Mock<UserProfileService>();
+            BucketerMock            = new Mock<Bucketer>(LoggerMock.Object);
             
-            ValidProjectConfig = ProjectConfig.Create(TestData.ValidDataFileV3, LoggerMock.Object, ErrorHandlerMock.Object);
-            ProjectConfig = ProjectConfig.Create(TestData.Datafile, LoggerMock.Object, ErrorHandlerMock.Object);
+            ValidProjectConfig      = ProjectConfig.Create(TestData.ValidDataFileV3, LoggerMock.Object, ErrorHandlerMock.Object);
             NoAudienceProjectConfig = ProjectConfig.Create(TestData.NoAudienceProjectConfigV3, LoggerMock.Object, ErrorHandlerMock.Object);
-            WhitelistedExperiment = ValidProjectConfig.ExperimentIdMap["223"];
-            WhitelistedVariation = WhitelistedExperiment.VariationKeyToVariationMap["vtag1"];
-            DecisionService = new DecisionService(new Bucketer(LoggerMock.Object), ErrorHandlerMock.Object, ProjectConfig, null, LoggerMock.Object);
+            WhitelistedExperiment   = ValidProjectConfig.ExperimentIdMap["223"];
+            WhitelistedVariation    = WhitelistedExperiment.VariationKeyToVariationMap["vtag1"];
+            ProjectConfig           = ProjectConfig.Create(TestData.Datafile, LoggerMock.Object, ErrorHandlerMock.Object);
+            DecisionService         = new DecisionService(new Bucketer(LoggerMock.Object), ErrorHandlerMock.Object, ProjectConfig, null, LoggerMock.Object);
 
-            DecisionServiceMock = new Mock<DecisionService>(BucketerMock.Object, ErrorHandlerMock.Object, ProjectConfig, null, LoggerMock.Object);
-            DecisionServiceMock.CallBase = true;
+            DecisionServiceMock     = new Mock<DecisionService>(BucketerMock.Object, ErrorHandlerMock.Object, ProjectConfig, null, LoggerMock.Object) { CallBase = true };
```

Comment:
why CallBase = true? what does it mean?

Reply:
"CallBase = true" allows Mock class to call base class methods. It is required in our unit tests as we are setting expectations on some methods that are going to be called from actual base class methods.

For detials, please read this:
https://stackoverflow.com/questions/7691796/how-to-use-moq-to-test-a-concrete-method-in-an-abstract-class